src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -527,6 +528,65 @@ public void WriteByteArray(byte[] bytes)
             Write(bytes);
         }
 
+        public async Task<bool> WriteConst(ConstantTypeCode? type, byte[] bytes, MonoSDBHelper SdbHelper, CancellationToken token)
+        {
+            switch (type)
+            {
+                case ConstantTypeCode.Invalid:
+                    return false;
+                case ConstantTypeCode.Boolean:
+                    var boolVal = BitConverter.ToBoolean(bytes);
+                    Write(ElementType.Boolean, boolVal ? 1 : 0);
+                    return true;
+                case ConstantTypeCode.Char:
+                    Write(ElementType.Char, (int)BitConverter.ToChar(bytes));
+                    return true;
+                case ConstantTypeCode.SByte:
+                    Write(ElementType.I1, (int)bytes[0]);
+                    return true;
+                case ConstantTypeCode.Byte:
+                    Write(ElementType.U1, (uint)bytes[0]);
+                    return true;
+                case ConstantTypeCode.Int16:
+                    Write(ElementType.I2, (uint)BitConverter.ToInt16(bytes, 0));
+                    return true;
+                case ConstantTypeCode.UInt16:
+                    Write(ElementType.U2, (uint)BitConverter.ToUInt16(bytes, 0));
+                    return true;
+                case ConstantTypeCode.Int32:
+                    Write(ElementType.I4, BitConverter.ToInt32(bytes, 0));
+                    return true;
+                case ConstantTypeCode.UInt32:
+                    Write(ElementType.U4, BitConverter.ToUInt32(bytes, 0));
+                    return true;
+                case ConstantTypeCode.Int64:
+                    Write(ElementType.I8, BitConverter.ToInt64(bytes, 0));
+                    return true;
+                case ConstantTypeCode.UInt64:
+                    Write(ElementType.U8, BitConverter.ToUInt64(bytes, 0));
+                    return true;
+                case ConstantTypeCode.Single:
+                    Write(ElementType.R4, BitConverter.ToSingle(bytes, 0));
+                    return true;
+                case ConstantTypeCode.Double:
+                    Write(ElementType.R8, BitConverter.ToDouble(bytes, 0));
+                    return true;
+                case ConstantTypeCode.String:
+                    // encoded chars are separated by trailing nulls: '\0' that indicate the end of string,
+                    // leaving them results in cutting the string after 1st char
+                    var stringVal = Encoding.UTF8.GetString(bytes).Replace("\0", "");
```

Comment:
These are going to be utf-16 strings so use
```suggestion
                    var stringVal = Encoding.Unicode.GetString(bytes);
```

Reply:
And add a test for such utf-16 strings ;)